ResearchKit/Common/ORKFormItemCell.m

```diff
@@ -469,26 +488,59 @@ - (void)answerDidChange {
     }
 }
 
+- (BOOL)isAnswerValid {
+    NSString *text = self.textField.text;
+    BOOL isValid = YES;
+    if ([text length]) {
```

Comment:
if `isAnswerValidWithString` has the guard for exclude zero length text, then this `if` can be removed.